server/controllers/categorised/employees.js

```diff
@@ -63,16 +99,125 @@ exports.checkHoliday = function (req, res, next) {
   .then(function (result) {
     res.send(result);
   })
-  .catch(function (err) { next(err); })
+  .catch(next)
   .done();
 };
 
+/**
+* Check an existing offday
+*/
 exports.checkOffday = function (req, res, next) {
-  var sql ='SELECT * FROM offday WHERE date = "' + req.query.date + '" AND id <> "' + req.query.id +'"';
-  db.exec(sql)
+  var sql ='SELECT * FROM offday WHERE date = ? AND id <> ?';
+  db.exec(sql, [req.query.date, req.query.id])
   .then(function (result) {
     res.send(result);
   })
-  .catch(function (err) { next(err); })
+  .catch(next)
+  .done();
+};
+
+/**
+* Returns an object of details of an employee referenced by an `id` in the database
+*
+* @param {object} request The express request object
+* @param {object} response The express response object
+* @param {object} next The express middleware next object
+*
+* @example
+* // GET /employees/:id : Get details of an employee
+* var employees = require('categorised/employees');
+* employees.details(request, response, next);
+*/
+exports.details = function (req, res, next) {
+  var sql =
+  'SELECT ' +
+  'employee.id, employee.code AS code_employee, employee.prenom, employee.name, ' +
+  'employee.postnom, employee.sexe, employee.dob, employee.date_embauche, employee.service_id, ' +
+  'employee.nb_spouse, employee.nb_enfant, employee.grade_id, employee.locked, grade.text, grade.basic_salary, ' +
+  'fonction.id AS fonction_id, fonction.fonction_txt, service.name AS service_txt, ' +
+  'employee.phone, employee.email, employee.adresse, employee.bank, employee.bank_account, employee.daily_salary, employee.location_id, ' +
+  'grade.code AS code_grade, debitor.uuid as debitor_uuid, debitor.text AS debitor_text,debitor.group_uuid as debitor_group_uuid, ' +
+  'creditor.uuid as creditor_uuid, creditor.text AS creditor_text, creditor.group_uuid as creditor_group_uuid, creditor_group.account_id ' +
+  'FROM employee ' +
+  ' JOIN grade ON employee.grade_id = grade.uuid ' +
+  ' JOIN fonction ON employee.fonction_id = fonction.id ' +
+  ' JOIN debitor ON employee.debitor_uuid = debitor.uuid ' +
+  ' JOIN creditor ON employee.creditor_uuid = creditor.uuid ' +
+  ' JOIN creditor_group ON creditor_group.uuid = creditor.group_uuid ' +
+  ' LEFT JOIN service ON service.id = employee.service_id ' +
+  'WHERE employee.id = ? ';
+
+  db.exec(sql, [req.params.id])
+  .then(function (rows) {
+    // send a 204 if rows is empty
+    if (!rows.length) {
+      return res.status(204).json({
```

Comment:
A more appropriate error code would be [404 NOT FOUND](https://en.wikipedia.org/wiki/HTTP_404).  It indicates that the specific ID was not found in the database.  2XX is a success code, but the server did not successfully find the information for the provided ID.